fast64_internal/oot/exporter/other/scene_table.py

```diff
@@ -0,0 +1,324 @@
+import os
+import enum
+import bpy
+
+from dataclasses import dataclass, field
+from typing import Optional, TYPE_CHECKING
+from ....utility import PluginError, writeFile
+from ...oot_constants import ootEnumSceneID, ootSceneNameToID
+from ...oot_utility import getCustomProperty, ExportInfo
+
+if TYPE_CHECKING:
+    from ..main import SceneExport
+
+
+ADDED_SCENES_COMMENT = "// Added scenes"
+
+
+class SceneIndexType(enum.IntEnum):
+    """Used to figure out the value of ``selectedSceneIndex``"""
+
+    # this is using negative numbers since this is used as a return type if the scene index wasn't found
+    CUSTOM = -1  # custom scene
+    VANILLA_REMOVED = -2  # vanilla scene that was removed, this is to know if it should insert an entry
+
+
+@dataclass
+class SceneTableEntry:
+    """Defines an entry of ``scene_table.h``"""
+
+    index: int
+    original: Optional[str]  # the original line from the parsed file
+    exporter: Optional["SceneExport"] = None
+    exportName: Optional[str] = None
+    prefix: Optional[str] = None  # ifdefs, endifs, comments etc, everything before the current entry
+    suffix: Optional[str] = None  # remaining data after the last entry
```

Comment:
I was able to remove the first two, but I still need the prefix/suffix because of HackerOoT (and future decomp update™) that have an ifdef to disable debug scenes